vm/src/exceptions.rs

```diff
@@ -1458,10 +1575,56 @@ pub(super) mod types {
     #[derive(Debug)]
     pub struct PyRecursionError {}
 
-    #[pyexception(name, base = "PyException", ctx = "syntax_error", impl)]
+    #[pyexception(name, base = "PyException", ctx = "syntax_error")]
     #[derive(Debug)]
     pub struct PySyntaxError {}
 
+    #[pyexception]
+    impl PySyntaxError {
+        #[pymethod(magic)]
+        fn str(exc: PyBaseExceptionRef, vm: &VirtualMachine) -> PyStrRef {
+            fn basename(filename: &str) -> &str {
+                // TODO: RUSTPYTHON use OS-dependent path separator as files like a\b.txt could exist in unix
+                filename.rsplit(&['/', '\\']).next().unwrap_or(filename)
```

Comment:
For this, I was thinking of using `#[cfg(target_family = ???)]` to determine the path separator. Would wasm be forward slash?

Reply:
I am pretty sure checking windows or not practically works.
```rust
let separator = if cfg!(windows) {
    '\\'
} else {
    '/'
};
```

Reply:
done